rollout/sync.go

```diff
@@ -698,16 +701,35 @@ func (c *rolloutContext) persistRolloutStatus(newStatus *v1alpha1.RolloutStatus)
 		c.requeueStuckRollout(*newStatus)
 		return nil
 	}
+
 	newRollout, err := c.argoprojclientset.ArgoprojV1alpha1().Rollouts(c.rollout.Namespace).Patch(ctx, c.rollout.Name, patchtypes.MergePatchType, patch, metav1.PatchOptions{}, "status")
 	if err != nil {
 		logCtx.Warningf("Error updating rollout: %v", err)
 		return err
 	}
+
+	c.sendStateChangeEvents(&prevStatus, newStatus)
 	logCtx.Infof("Patched: %s", patch)
 	c.newRollout = newRollout
 	return nil
 }
 
+// sendStateChangeEvents emit rollout events on significant state changes
+func (c *rolloutContext) sendStateChangeEvents(prevStatus, newStatus *v1alpha1.RolloutStatus) {
+	prevPaused := len(prevStatus.PauseConditions) > 0
```

Comment:
i see in lot of places we check for len of pausedconditions..is there a case where there can be more than one pause conditions applies?

Reply:
Good question. A background analysis run could pause a canary rollout because of inconclusive analysis, but a rollout could also be automatically paused from a pause step. If the pause step had a duration, we want to automatically clear the pause step (by removing that condition), but leave the pause condition for inconclusive analysis.